### ISSUE TITLE
Reduce footprint of power cables/lines without voltage

### ISSUE THREAD

**Issue by gustavecha** [gravitystorm/openstreetmap-carto#3460]
### Expected behavior
Power cables (and lines/minor_lines) rendering should adapt on voltage=* value if set.
overground cables and lines/minor_line (and their supports if possible) with no voltage or voltage < 1000 should be rendered only at zoom 18 to 20 with light line.

Tagging mailing list agree on downgrading the render of lines/cables with no voltage or voltage < 1000
As a side effect, this will encourage users to set voltage also.

### Actual behavior
Currently, there are power cables shown on the map that appear massive while they are only intended for local distribution of power. It clutters the map.
Voltage isn't used to adapt render of lines/cable.

### Links and screenshots illustrating the problem
See https://www.openstreetmap.org/way/512818816 which is a 400v cable rendered the same as a 20 000v cable

![screenshot_2018-10-20 linia 512818816 openstreetmap](https://user-images.githubusercontent.com/5439713/47250345-f9833380-d41f-11e8-9170-d573e96c7bad.png)

**Comment by matkoniecz** [comment]
> https://taginfo.openstreetmap.org/tags/power=line#combinations

1/3 of power=line has no voltage tag set. One should not demand from mapper to add this highly detailed information for basic rendering.

Doing that will probably result in adding guessed or completely false `voltage` tags.

**Comment by gustavecha** [comment]
> See https://taginfo.openstreetmap.org/tags/power=line#combinations
> 
> 1/3 of power=line has no voltage tag set. One should not demand from mapper to add this highly detailed information for basic rendering.
> 
> Doing that will probably result in adding guessed or completely false `voltage` tags.

Encouraging people to contribute to OSM may result in false tagging and mistakes.
In pratice OSM works well despite vandalism.
Your whole work on carto encourage people to use some tags more than others because it makes things appear on map.

> I agree with @matkoniecz. Also, just because it doesn't have a voltage tag, doesn't automatically make it a minor power line. 

That wasn't my point.
Because lines aren't well described, their render should be less prominent and took the minor_line thin line as a possibility. No problem to make a difference between actual minor and less described lines.
It's exactly the same that assuming all building=yes are houses just because it's obvious a building is a house. Not at all and we should produce tools that encourage people to provide more objective information, not less.

> A lot of major power lines in California are missing the voltage tag, but its obvious they are major power lines by the types of towers they are attached to. Rendering them as minor lines would essentially strip them of proper rendering, simply because a tag is missing that isn't required to be added to the object in the first place.

What I see here : https://openinframap.org/#5.93/35.055/-116.515 is that many major lines are properly rendered since the voltage is actually known.
If nothing encourage to, who will make efforts to add missing ones?
Voltages are *written* on each tower or poles by operators. Is this so hard to go there and read?

> That is clearly the wrong way to go about it for the reasons @matkoniecz gave. At least in this situation.

Sorry guys, as always I look for solutions to encourage people to look for additional data that makes OSM better.
As many times before, folks are here to say how wrong this is but without any suggestions of propositions. That's not really kind.

**Comment by matkoniecz** [comment]
>  Your whole work on carto encourage people to use some tags more than others because it makes things appear on map.

The trick is to strongly encourage providing information that is nearly always accessible and reasonable to tag for a normal person. I feel that "power lines lines will not render properly without `voltage` tag" is not one of cases that fit.

> Because lines aren't well described, their render should be less prominent

I think that this is a fundamental difference. You consider `power=line` without `voltage` tag to not be well tagged like `highway=road`. While I think that `power=line`, `power=minor_line` distinction is sufficient for general purpose map not focused on power networks.

> Voltages are written on each tower or poles by operators. Is this so hard to go there and read?

It is hard to read labels when power towers are hundreds of meters away, with no path access to them. And it is even harder to read labels on power towers from aerial images.

Also, note that just because your region has labels like that it is not guaranteed that it is the same in other countries.

**Comment by matkoniecz** [comment]
> Currently, there are power cables shown on the map that appear massive while they are only intended for local distribution of power.

Wiki has at https://wiki.openstreetmap.org/wiki/Tag:power=cable?uselang=en

> These heavier cables should not be confused with light-weight uninsulated power distribution lines supported by pylons and poles, for which power=line/power=minor_line are normally more appropriate.

What is wrong? Wiki? This tagging example? In general it seems like a poor tagging scheme, how normal person is supposed to recognize whatever given line is insulated or not?

**Comment by matkoniecz** [comment]
I created https://wiki.openstreetmap.org/wiki/Talk:Tag:power%3Dcable#how_normal_person_is_supposed_to_guess_whatever_given_overhead_line_is_power.3Dcable_or_power.3Dline.3F because maybe I am confused and this tagging scheme makes some sort of sense.

**Comment by jeisenbe** [comment]
I agree that we shouldn’t change the rendering of power=line based on
voltage or lack thereof.

However, I do think it would be good to move low-voltage power=minor_line
to a higher zoom level. This way, main distribution lines in rural areas
would still be visible at zoom 16, but we could reduce the number of
low-voltage lines shown in residential areas of cities.

>=1000volts is the standard for high voltage, so anything under that could
start at z18

**Comment by dieterdreist** [comment]
Am Fr., 19. Okt. 2018 um 17:40 Uhr schrieb Gustave <notifications@github.com
>:

> Expected behavior
>
> Power cables (and lines/minor_lines) rendering should adapt on voltage=*
> value if set.
> overground cables and lines/minor_line (and their supports if possible)
> with no voltage or voltage < 1000 should be rendered only at zoom 18 to 20
> with light line.
>
> Tagging mailing list agree on downgrading the render of lines/cables with
> no voltage or voltage < 1000
> As a side effect, this will encourage users to set voltage also.
> Actual behavior
>
> Currently, there are power cables shown on the map that appear massive
> while they are only intended for local distribution of power. It clutters
> the map.
> Voltage isn't used to adapt render of lines/cable.
>


no idea about cables but for major power lines it does not seem like a good
idea to push them to zoom 18. Also for minor lines this is late. The idea
of showing them in this style is mainly for orientation.

**Comment by jeisenbe** [comment]
1. I checked out the possibility of using voltage < 1000 to render power=minor_line at z18 instead of z16, however, I noticed that the power poles still render at z16. Because these are separately rendered as nodes tagged power=pole, it would be difficult to avoid rendering the lines at z16 and z17. 

I don't know if there is a way to merge tags from a way onto it's nodes?

```
#power-poles {
  [zoom >= 16] {
    marker-file: url('symbols/square.svg');
    marker-fill: #928f8f;
    marker-width: 3;
  }
```

2. It would be possible to render power=cable like power=minor_line at lower voltage. But what would be the voltage cut-off? 

I think it would be nice if there were a tag power=minor_cable for symmetry, and because the huge insulated cables used on very high-voltage transmission lines are quite different from the small insulated wires carrying 220 volts to a house from the nearest minor line.

**Comment by kocio-pl** [comment]
power=minor_cable is both very rare and has no wiki page at the moment:

![taghistory 17](https://user-images.githubusercontent.com/5439713/47365163-42b2dc00-d6db-11e8-9beb-4a19668bf7ad.png)

**Comment by matkoniecz** [comment]
@gustavecha 

Can you consider replying at https://wiki.openstreetmap.org/wiki/Talk:Tag:power%3Dcable#how_normal_person_is_supposed_to_guess_whatever_given_overhead_line_is_power.3Dcable_or_power.3Dline.3F because maybe I am confused and this tagging scheme makes some sort of sense.

Also, I suspect that problem is caused by wrong tagging:

> Currently, there are power cables shown on the map that appear massive while they are only intended for local distribution of power.

Wiki has at https://wiki.openstreetmap.org/wiki/Tag:power=cable?uselang=en

> These heavier cables should not be confused with light-weight uninsulated power distribution lines supported by pylons and poles, for which power=line/power=minor_line are normally more appropriate.

What is wrong? Wiki? This tagging example? In general it seems like a poor tagging scheme, how normal person is supposed to recognize whatever given line is insulated or not?

---------

In short: I suspect that problem is caused by a fundamentally broken tagging scheme (insulation selected as fundamental property that divides `power=cable` and `power=line/minor_line`) or incorrect tagging is a real problem, not rendering. And `voltage` rendering as proposed here would hide real problems at cost of making properly tagged objects showed in substandard way.

**Comment by matkoniecz** [comment]
@gustavecha and anybody else interested in electricity network topic - ping again - can you reply to https://github.com/gravitystorm/openstreetmap-carto/issues/3460#issuecomment-432640879 ? Currently I am planning to close it as wrong data causing wrong rendering based on https://wiki.openstreetmap.org/wiki/Tag:power=cable (with bonus complaint about badly-designed tagging scheme putting insulation as the most important property).

**Comment by gustavecha** [comment]
Fanfouer's answer is acceptable.
Ends are good place to see if the coating is an actual insulation or not

**Comment by matkoniecz** [comment]
What about someone mapping from aerial images or in case of power wire on a big tower or not standing next to a pole?

**Comment by gustavecha** [comment]
Aerial view doesn't allow to know this. Unfortunately.
In France we have full public data available regarding line vs cable on transmission network. Distribution network is solved with mapillary or survey
That's why i'd really prefer people use power=line everytime and insulation come with additional tags

Regarding #3460 no, currently, if it's insulated => power=cable and if not power=line. There is no minor_cable (and shouldn't) so no tagging mistake
Minor vs major is pointless

**Comment by matkoniecz** [comment]
There really should be insulated=yes/no tag, that would allow to tag properly also in regions without open data and to avoid this particular problem (with power=line/minor_line sufficient for orientation purposes).

**Comment by gustavecha** [comment]
There is actually an insulation proposal
https://wiki.openstreetmap.org/wiki/Proposed_features/Insulation_proposal

I agree on insulation, like voltage or material of conductors if applicable.

**Comment by matkoniecz** [comment]
Inspired by tagging issues that I discovered here I created my own https://wiki.openstreetmap.org/wiki/Proposed_features/isolated intended to be as simple as possible.

**Comment by matkoniecz** [comment]
It will be posted soon on tagging mailing list (please do not discuss it in this issue but rather on tagging mailing list and https://wiki.openstreetmap.org/w/index.php?title=Talk:Proposed_features/isolated&action=edit&redlink=1

**Comment by matkoniecz** [comment]
And going back to originally reported proposal:

> Reduce footprint of power cables/lines without voltage

I am going to decline it as only small part of power lines have voltage tagged. Maybe some other ideas floated here are feasible. One issues may be opened if this ideas were not proposed already.

See https://taginfo.openstreetmap.org/tags/power=line#combinations

1/3 of power=line has no voltage tag set. One should not demand from mapper to add this highly detailed information for basic rendering.
